src/activity_handlers.cpp

```diff
@@ -1943,7 +1943,7 @@ void activity_handlers::butcher_do_turn( player_activity *, player *p )
 
 void activity_handlers::read_finish( player_activity *act, player *p )
 {
-    p->do_read( act->targets[0].get_item() );
+    p->do_read( *act->targets[0].get_item() );
```

Comment:
Since you've touched this line, it would be good to replace `targets[0]` with `targets.front()` (and probably add an `assert` to make sure that it can actually be referenced).

Reply:
I changed the call to `front()`.

Regarding the `assert`, I'm going to add a function to `item_location` that returns an item *reference*, and that function will do its own assertion (but probably it will issue a debug message and throw an exception). Many places in the code base simply expect a valid `item_location`, the assertion should not be duplicated in all those places.